Runtime/Reporter/ReportUploadGuardService.cs

```diff
@@ -0,0 +1,37 @@
+using BugSplatUnity.Runtime.Settings;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+
+namespace BugSplatUnity.Runtime.Reporter
+{
+    internal interface IReportUploadGuardService
+    {
+        public bool ShouldPostLogMessage(IClientSettingsRepository clientSettingsRepository, Exception exception, LogType type);
+        public bool ShouldPostException(IClientSettingsRepository clientSettingsRepository, Exception exception);
+    }
+
+    internal class ReportUploadGuardService : IReportUploadGuardService
+    {
+        public bool ShouldPostException(IClientSettingsRepository clientSettingsRepository, Exception exception)
+        {
+            var shouldPostException = clientSettingsRepository.ShouldPostException(exception);
+
+            if (Application.isEditor)
+            {
+                shouldPostException &= clientSettingsRepository.PostExceptionsInEditor;
+            }
+
+            return shouldPostException;
+        }
+
+        public bool ShouldPostLogMessage(IClientSettingsRepository clientSettingsRepository, Exception exception, LogType type)
```

Comment:
As far as I can tell, this is never called with the Exception parameter. Is that true? If so can we remove this parameter?

Reply:
Yeah, it is only called with exception == null. I'll remove the parameter 👍